cli/deployment/config.go

```diff
@@ -393,21 +392,26 @@ func setConfig(prefix string, vip *viper.Viper, target interface{}) {
 		typ = val.Type()
 	}
 
-	// Manually bind to env to support CODER_$INDEX_$FIELD format for structured slices.
-	_ = vip.BindEnv(prefix, formatEnv(prefix))
-
+	// Ensure that we only bind env variables to proper fields,
+	// otherwise Viper will get confused if the parent struct is
+	// assigned a value.
```

Comment:
Thanks for writing the comment here. I'm afraid that we may run into this issue again in the future (with library bump). Do you think that it's possible to prepare a basic unit test to validate the behavior?

Reply:
I'm afraid [`2f4ac30` (#4893)](https://github.com/coder/coder/pull/4893/commits/2f4ac3013ebc62dc123e79bb91afba2fbc498099) is the best I can do. I didn't find too many ways to reproduce the issue.